src/bootstrap/flags.rs

```diff
@@ -232,7 +232,12 @@ To learn more about a subcommand, run `./x.py <subcommand> -h`",
         match subcommand.as_str() {
             "test" | "t" => {
                 opts.optflag("", "no-fail-fast", "Run all tests regardless of failure");
-                opts.optmulti("", "test-args", "extra arguments", "ARGS");
+                opts.optmulti(
+                    "",
+                    "test-args",
+                    "extra arguments to pass to cargo when running tests",
```

Comment:
```suggestion
                    "extra arguments to pass to the test binary when running tests",
```
That covers all the bases I think: cargo, libtest, and rustdoc. Maybe also name those three in the help message?

Reply:
I don't know if we really need `compiletest, rustdoc, libtest` in the phrasing, but I might be misunderstanding your suggestion.

When you say "...in the help message" you mean:
1 - have those in the phrase like `extra arguments to pass to the test binary when running tests (ex. libtest, compiletest or rustdoc)` ?
2 - or eventually to add a new paragraph in the `Arguments` section of the `x.py test --help` output?

atm I think that adding the tools name might be unnecessary, and after the clarification (to me) in https://github.com/rust-lang/rust/pull/78243#issuecomment-714688574 , I believe that something along the lines `Arguments to be passed for the test tool being used (ex. libtest, compiletest or rustdoc)`  would suffice.

but then again, can you clarify my questions and if possible provide example?

Reply:
All your suggestions seem reasonable to me, the line
> Arguments to be passed for the test tool being used (ex. libtest, compiletest or rustdoc)
is about what I was thinking of.

Reply:
ok great, so changing the description to:
```suggestion
                    "Arguments to be passed for the test tool being used (ex. libtest, compiletest or rustdoc)",
```
would suffice and we can call it a day? :)

Reply:
nit: 
```suggestion
                    "extra arguments to be passed for the test tool being used (e.g. libtest, compiletest or rustdoc)
```
but otherwise yes, that seems fine.